pyfactorybridge/__init__.py

```diff
@@ -121,54 +120,125 @@ def __request(
         except (ConnectionError, Timeout):
             raise ServerError("The server could not be reached")
 
-    def authorise_password(self, Password) -> BearerAuth | None:
+    def renew_auth(
+        self,
+        method: str | None = None,
+        value: str | None = None,
+        permissions: str = "Administrator",
+    ):
+        """Renew the authentication token based on the provided method."""
+
+        if permissions not in [
+            "Administrator",
+            "Client",
+            "APIToken",
+            "NotAuthenticated",
+        ]:
```

Comment:
Maybe this could be handled a bit more elegantly using enums instead

Reply:
I don't think an enum would be useful here